scripts/check-legend-application-documentation-broken-links.js

```diff
@@ -0,0 +1,81 @@
+const fs = require("fs");
```

Comment:
I generally agree with the logic of this file, but there are a few problems:
1. We're using fairly old JS syntax
2. The await/async logic seems odd and could be buggy so beware
3. The file name/file path resolution logic seems unnecessarily convoluted
4. We could improve support for external doc links: e.g. youtube, github wiki (with some anchors)